src/System.IO/tests/BinaryWriter/BinaryWriter_WriteByteCharTests.cs

```diff
@@ -0,0 +1,571 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.IO;
+using System.Text;
+
+namespace BinaryWriterTests
+{
+    public class BinaryWriter_WriteByteCharTests
+    {
+        /// <summary>
+        /// Cases Tested:
+        /// 1) Tests that BinaryWriter properly writes chars into a stream.
+        /// 2) Tests that if someone writes surrogate characters, an argument exception is thrown
+        /// 3) Casting an int to char and writing it, works.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteCharTest()
```

Comment:
Also should be split up. I'll hold off on more comments about splitting because it just occurred to me that these are ported and not your tests :) I'll just look for potential errors.